### PR TITLE
Ug feat size optimization

### PR DESCRIPTION
**Pull request by ujjwalgupta94**
This PR reduces the size of bundled files being export with some other fixes listed below - 

1. Reduce bundle size
2. Create dev & env bundle and export them accordingly
3. remove bluebird.js & implement method mapPromise
4. moves dev dependencies like typings to dev dependencies from dependecies
5. create script for dev & prod build
6. move typings from lib to dist/ts

once this PR is merged, please run `npm run deploy` to build and publish.

### CODE REVIEW

**Comment by ankitboghra** [comment]
Suggestion: can replace `axios`(6.6K) with fetch(`node-fetch`)(1.3k)

**Comment by jdkanani** [comment]
@ankitboghra @ujjwalgupta94 We faced few issues with node-fetch. @rahuldamodar94 can tell more about it. But we can look deeper.

**Review by jdkanani — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ bundle.js
 dist
 
 
+*.tgz
```
[the remark this answers is not in the document]
Comment (ujjwalgupta94) [reply .gitignore]: done

File: src/libs/ProofsUtil.ts [hunk 2]
```diff
@@ -149,14 +149,14 @@ export default class ProofsUtil {
...
+    await mapPromise(
       headers,
       // eslint-disable-next-line
       async (_, i) => {
         logger.debug('fetching block', i + start)
         headers[i] = ProofsUtil.getBlockHeader(await web3.eth.getBlock(i + start))
-      },
-      { concurrency: 20 }
+      }
+      // { concurrency: 20 }
```
[the remark this answers is not in the document]
Comment (ujjwalgupta94) [reply src/libs/ProofsUtil.ts]: added concurrency back with concurrency implementation in mapPromise

**Review by jdkanani — APPROVED** [review]

**Pull request merged**